job-service-db/src/main/resources/procedures/processFailedDependentJobs.sql

```diff
@@ -36,13 +36,14 @@ BEGIN
         WITH RECURSIVE all_job_dependencies AS (
             SELECT partition_id, dependent_job_id, job_id
             FROM job_dependency
-            UNION ALL
+            WHERE partition_id = in_partition_id AND job_id = in_job_id
+            UNION
```

Comment:
Why this this moving to `UNION`?  Surely that is a more expensive operator than `UNION ALL`?

Reply:
Using `UNION` drastically reduced the number of iterations the loop does. To the point where `UNION ALL` would used 100s of GB of disk and not finish while `UNION` was sub-second.